### PR TITLE
lsvmbus case enhancement

### PR DESCRIPTION
**Pull request by LiliDeng**
- Update case lsvmbus_channel_counting into lsvmbus_count_devices_channels.
- Enhance name_convention.
- Make assert message clear.

### CODE REVIEW

File: lisa/tools/lsvmbus.py [hunk 1]
```diff
@@ -172,8 +172,10 @@ def install(self) -> bool:
...
 
-    def get_vmbuses(self, force_run: bool = False) -> List[VmBus]:
-        if (not self._vmbuses) or force_run:
+    def get_device_channels_from_lsvmbus(
+        self, force_run: bool = False
+    ) -> List[VmBusDevice]:
+        if (not self._vmbus_devices) or force_run:
             result = self.run("-vv", force_run=force_run, shell=True)
             if result.exit_code != 0:
                 result = self.run("-vv", shell=True, sudo=True)
```
Comment (squirrelsc) [inline lisa/tools/lsvmbus.py]: pass force_run to this call too.

File: testsuites/basic/lsvmbus.py [hunk 2]
```diff
@@ -47,42 +46,46 @@ class LsVmBus(TestSuite):
...
-        for vmbus in vmbus_list:
-            if vmbus.vmbus_name == "Synthetic network adapter":
-                assert_that(vmbus.channel_vp_map).is_length(
-                    expected_network_channel_count
-                )
-            if vmbus.vmbus_name == "Synthetic SCSI Controller":
-                assert_that(vmbus.channel_vp_map).is_length(expected_scsi_channel_count)
+        for vmbus_device in vmbus_devices_list:
+            if vmbus_device.name == "Synthetic network adapter":
+                assert_that(
```
Comment (squirrelsc) [inline testsuites/basic/lsvmbus.py]: The original assertion would be better. If you want to refine the output of `vmbus_device.channel_vp_map`, it needs to implement `__str__` of `ChannelVPMap`.

Comment (LiliDeng) [reply testsuites/basic/lsvmbus.py]: vmbus.channel_vp_map is list of ChannelVPMap, I implemented __str__ of ChannelVPMap, but only vmbus.channel_vp_map[0].__str__() print expected value.

Comment (squirrelsc) [reply testsuites/basic/lsvmbus.py]: Implement the `__str__`, you can use `f"{vmbus_device.channel_vp_map}"`. If you need to `log.info(vmbus_device.channel_vp_map)`, it needs to implement `__repr__`. You can implement them both.

File: testsuites/basic/lsvmbus.py [hunk 3]
```diff
@@ -47,42 +46,46 @@ class LsVmBus(TestSuite):
...
             - Synthetic SCSI Controller
             - Synthetic IDE Controller (gen1 only)
-        2. check channel counts of each netvsc and storvsc SCSI device.
-        3. check expected channel count of each netvsc is min (num of vcpu, 8).
-        4. check expected channel count of each storvsc SCSI device is min
-            (num of vcpu/4, 64).
-        from the output of lsvmbus -vv command.
+        2. Check that each netvsc and storvsc SCSI device have correct number of vmbus
+            channels created and associated.
+            2.1 Check expected channel count of each netvsc is min (num of vcpu, 8).
```
Comment (glima) [inline testsuites/basic/lsvmbus.py:51]: There was some suggestion of pointing to kernel's file:line to prove/back this here, no?

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**